### PR TITLE
fix bug

### PR DESCRIPTION
**Pull request by augustinanders**
fixed a bug, before it was possible to doubleclick the backside of a card. This caused a problem, because the same image was put to the `compareImages`-state twice. My solution is a Timeout with the same duration of the css-animation(600ms).

Edit:
Refactored a lot of code: maybe review in `codespaces` to get a better overview.

### CODE REVIEW

**Review by toshydev — APPROVED** [review]
LGTM ✅
Great fix. I did some QA as well but this fix might need several QAs just to be sure.
Amazing work!💪

File: components/Navbar/index.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  }
+`;
+export default function Navbar() {
+  const router = useRouter();
+
+  return (
+    <StyledNavbar>
+      <StyledList>
+        <Link href="/">
+          <StyledListItem isActive={router.pathname === "/"}>🎮</StyledListItem>
```
Comment (PReichetanz) [inline components/Navbar/index.js:44]: There should be an `aria-label` to announce the correct behaviour to a screen reader as well.

File: zustand/useHighscoresStore.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+// A function that returns a custom hook that can be used to access the state.
+function createLocalStorageStore(initialStore, name) {
+  // Create a Zustand store without persistence.
+  const useServerStore = create(initialStore);
+  // Create a Zustand store with persistence to local storage.
+  const useClientStore = create(persist(initialStore, { name }));
+
+  // A custom hook that selects the appropriate store based on whether
+  // the component is hydrated or not.
```
Comment (PReichetanz) [inline zustand/useHighscoresStore.js:28]: Do you still need these comments in production?

**Review by PReichetanz — COMMENTED** [review]
Nice work! Left only some minor comments to discuss.

**Comment by augustinanders** [comment]
> Nice work! Left only some minor comments to discuss.

Thank you for the review! changed the suggestions. @PReichetanz

**Pull request merged**